cartography/intel/crxcavator.py

```diff
@@ -0,0 +1,210 @@
+import logging
+import json
+import requests
+import requests.auth
+import os
+
+from cartography.util import run_cleanup_job
+
+logger = logging.getLogger(__name__)
+
+# API key for CRXcavator generaated from crxcavator.io portal
+CRXcavator_API_KEY = os.environ.get('CREDENTIALS_CRXCAVATOR_API_KEY')
+
+# API for the CRXcavator API - https://api.crxcavator.io/v1 as of 07/09/19
+CRXcavator_API_BASE_URL = os.environ.get('CRXCAVATOR_URL')
+
+
+def get_extensions():
+    """
+    Get all of the installed extension metadata
+    :return: JSON text blob containing all extension metadata defined at
+    https://crxcavator.io/apidocs#tag/group/paths/~1group~1extensions~1combined/get
+    """
+    return call_crxcavator_api("/group/extensions/combined")
+
+
+def get_users_extensions():
+    """
+    Gets listing of all users who have installed each extension
+    :return: JSON text blob containing user email to extension id mapping defined at
+    https://crxcavator.io/apidocs#tag/group/paths/~1group~1users~1extensions/get
+    """
+    return call_crxcavator_api("/group/users/extensions")
+
+
+def call_crxcavator_api(api_and_parameters):
+    """
+    Perform the call requested to the CRXcavator API
+    :param api_and_parameters: Query string for the API including the required parameters
+    :return: Returns JSON text blob for the API called. API spec is at https://crxcavator.io/apidocs
+    """
+    uri = CRXcavator_API_BASE_URL + api_and_parameters
+    data = requests.get(
+        uri,
+        headers={'Accept': 'application/json',
+                 'API-Key': CRXcavator_API_KEY})
+    if data.ok:
+        return data.json()
+    else:
+        logger.error('Error during CRXcavator API call {}'.format(data.status_code))
+        return None
+
+
+def ingest_extensions(session, update_tag):
+    """
+    Ingests the extension details into Neo4J
+    :param session: Neo4J session object for server communication
+    :param update_tag: Timestamp used to determine data freshness
+    :return: None
+    """
+    # the JSON returned from the CRXcavator API does not return well formatted objects
+    # instead, each object is named after it's key, making enumeration more difficult
+    # will build a cleaner object for import into graph
+
+    _data_index = 1
```

Comment:
Is this needed?  `ingest_users_extensions()` below doesn't use variables for row/col indexing.

Reply:
There is a big difference in the way the data is returned between the two API calls. See the data samples added to tests. Open to suggestions for refactoring if there is an better method to clean this data up.

Reply:
Gotcha; was just wondering why you chose to have a var name for the indexes in this function but not the other.  Yeah, having the sample data helps.